### PR TITLE
Using JWTs and cookies for authentication

### PR DESCRIPTION
**Pull request by michellesfc**

### CODE REVIEW

File: fullstack_template/server/server.py [hunk 1]
```diff
@@ -15,14 +18,29 @@ def home():
...
         request_data = request.get_json()
         data = User(first_name=request_data["first_name"],
                     last_name=request_data["last_name"], 
                     email=request_data["email"],
-                    password=hash_password(request_data["password"]))
+                    password=hash_password(request_data["password"]),
+                    major=None)
+        user = User.query.filter_by(email=data.email).first()
+        if user is not None: 
+            return "unsucessful register"
```
Comment (AndyRewLy) [inline fullstack_template/server/server.py]: Can we send a json error message from this?  Can we send a response of 400 and specify that an email already exists?

Comment (AndyRewLy) [reply fullstack_template/server/server.py]: Same applies to the else case

File: fullstack_template/static/js/Login.jsx [hunk 2]
```diff
@@ -43,27 +43,28 @@ class Login extends Component {
...
-                        }
-                        else {
-				app.handleLogin("")
-                        }
-                })
+		}).then(function (data) {
+			const json = data;
+			console.log(json)
+			//storing JWT to a cookie
+			document.cookie = "access_token=" + json["access_token"]
```
Comment (AndyRewLy) [inline fullstack_template/static/js/Login.jsx]: Just wondering but is this cookie accessible throughout the user's whole session?

Comment (michellesfc) [reply fullstack_template/static/js/Login.jsx]: Yes, it should be accessible throughout the whole session.

File: fullstack_template/server/server.py (hunk 1 above)
Comment (michellesfc) [reply fullstack_template/server/server.py]: Yes, that is a good idea. I will make the changes.

File: fullstack_template/server/server.py [hunk 3]
```diff
@@ -31,25 +33,25 @@ def identity(payload):
...
+    data = User(first_name=request_data["first_name"],
+                last_name=request_data["last_name"], 
+                email=request_data["email"],
+                password=hash_password(request_data["password"]),
+                major=None)
+        
+    #email validation
+    user = User.query.filter_by(email=data.email).first()
+    if user is not None: 
+        return jsonify(message="Email already registered"), 400
```
Comment (AndyRewLy) [inline fullstack_template/server/server.py]: Does this 400 here return a status of 400?

Comment (michellesfc) [reply fullstack_template/server/server.py]: Yep

**Comment by michellesfc** [comment]
closes #12

**Pull request merged**